Server cache

iker-barriocanal opened the pull request:
# Cache implementation
The type of cache that has been implemented is the LFU (Least Frequently Used). You can learn about [cache replacement policies](https://en.wikipedia.org/wiki/Cache_replacement_policies) and the [LFU cache](https://en.wikipedia.org/wiki/Least_frequently_used).

## File summary
`server.data.cache` contains `Cache` and `Node` classes:
- `Cache` is a simple implementation of a LFU cache.
- `Node` is a node class used in `Cache`, in order to build a LinkedList of the most frequently used elements in the cache.

## Functionality
This is the functionality `Cache` implements:
- Constructors: a maximum number of elements may be given; but if not, it uses the default value (1000).
- `get` returns the mapping value of the cache from the given key, or `null` if there isn't stored such key.
- `contains` returns a boolean whether the given key is currently stored.
- `remove` deletes from the cache the key-value pair related to the given key.
- `clear` removes all the elements in the cache.
- Getters for the maximum number of elements and the current number of elements.

`toString` has been overriden so that returns a `String` containing all the key-value pairs stored in the cache, with the following format:
- If the cache is empty: `Cache (0/<maximumNumberelements>)`
- If the cache is not empty: `Cache (<currentNumberElements>/<maximumNumberelements>) - [<key1>:<value1>][<key2>:<value2>]`...

iker-barriocanal committed:
The node of the cache

iker-barriocanal committed:
Server cache

iker-barriocanal committed:
Cache finished with documentation

iker-barriocanal committed:
Merge branch 'master' into cache

iker-barriocanal committed:
Added functionality to Cache

`contains` and `remove` methods have been implemented

iker-barriocanal committed:
Cache integration in DAOs

iker-barriocanal committed:
Server cache testing

iker-barriocanal committed:
Improved cache test

Test coverage from 66% to 100%

iker-barriocanal committed:
Merge master into cache

iapellaniz commented on src/test/java/es/deusto/spq/server/data/dao/RoomDAOTest.java:
```diff
@@ -69,8 +69,15 @@ public void test2GetRoomByHotelId() {
-			Room room = roomDAO.getRoomById("R10");
-			Assert.assertEquals(rooms.get(0).getRoomId(), room.getRoomId());
+			/*
+			 * This test doesn't make any sense. There's a RoomDAO that it's not used to store rooms, but to retrieve them.
```
Then we've got to remove it

iapellaniz reviewed (changes requested):
Again, good! 

BTW (you don't need to change anything, it's just a small thing), JavaDoc comments' first line should only contain `/**`, so I recommend you start the text in a new line.

Just remove the test and we'll merge it right away.

iker-barriocanal committed:
Minor changes

Cache's and Node's documentation start in a new empty line.
Removed unused test in RoomDAOTest

iapellaniz committed:
Merge branch 'master' into cache
